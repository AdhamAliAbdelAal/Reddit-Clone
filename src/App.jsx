import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { CssBaseline, ThemeProvider } from '@mui/material';

import MainNavBar from './components/MainNavBar/MainNavBar';
import SettingsProfile from './components/Settings/SettingsProfile/SettingsProfile';
import SettingsPrivacy from './components/Settings/SettingsPrivacy/SettingsPrivacy';
import SettingsFeed from './components/Settings/SettingsFeed/SettingsFeed';
import SettingsAccount from './components/Settings/SettingsAccount/SettingsAccount';
import Settings from './pages/Settings';

import SignUpPage from './pages/SignUpPage';
import LogInPage from './pages/LogInPage';
import ForgetUsernamePage from './pages/ForgetUsernamePage';
import ForgetPasswordPage from './pages/ForgetPasswordPage';
import ResetPasswordPage from './pages/ResetPasswordPage';

import HomePage from './pages/HomePage';
import Messages from './pages/Messages';
import Moderation from './pages/Moderation';
import Profile from './pages/Profile';
import Search from './pages/Search';
import theme from './styles/theme';
import CreatePost from './pages/CreatePost';
import PostTypeContextProvider from './contexts/PostTypeContext';
import Notifications from './pages/Notifications';
import SubReddit from './pages/SubReddit';

function App() {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <PostTypeContextProvider>
        <Router className="App">
          <MainNavBar />
          <Routes>
            <Route
              path="/"
              element={
                <HomePage data-testid="home page" />
          }
            />
            <Route
              path="/:postClass"
              element={
                <HomePage />
          }
            />
            <Route
              path="/user"
              element={
                <Profile />
              }
            />
            <Route
              path="/register"
              element={
                <SignUpPage />
}
            />
            <Route
              path="/user/:username/"
              element={
                <Profile />
          }
            >
              <Route
                path=":subTitle"
                element={
                  <Profile />
          }
              >
                <Route
                  path=":sort"
                  element={
                    <Profile />
          }
                />

              </Route>
            </Route>

            <Route
              path="/login"
              element={
                <LogInPage />
          }
            />
            <Route
              path="/password"
              element={
                <ForgetPasswordPage />
          }
            />
            <Route
              path="/resetpassword/:token"
              element={
                <ResetPasswordPage />
          }
            />

            <Route
              path="/username"
              element={
                <ForgetUsernamePage />
          }
            />
            <Route
              path="/search"
              element={
                <Search />
          }
            />
            <Route
              path="/messages"
              element={
                <Messages />
          }
            />
            <Route
              path="/moderation"
              element={
                <Moderation />
          }
            />
            <Route
              path="/submit"
              element={
                <CreatePost />
            }
            />
            <Route
              path="/submit/:subredditName"
              element={
                <CreatePost />
            }
            />
            <Route
              path="/notifications"
              element={
                <Notifications />
          }
            />
            <Route
              path="/SubReddit/:Name"
              element={
                <SubReddit />
          }
            >
              <Route
                path=":postClass"
                element={
                  <SubReddit />
          }
              />
            </Route>
            <Route
              path="/settings"
              element={
                <Settings />
          }
            >
              <Route
                path=""
                element={
                  <SettingsAccount />
          }
              />
              <Route
                path="account"
                element={
                  <SettingsAccount />
          }
              />
              <Route
                path="profile"
                element={
                  <SettingsProfile />
          }
              />
              <Route
                path="privacy"
                element={
                  <SettingsPrivacy />
          }
              />
              <Route
                path="feed"
                element={
                  <SettingsFeed />
          }
              />
            </Route>
          </Routes>
        </Router>
      </PostTypeContextProvider>
    </ThemeProvider>
  );
}

export default App;
