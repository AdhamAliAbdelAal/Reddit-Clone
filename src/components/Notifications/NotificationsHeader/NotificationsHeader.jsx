import SettingsOutlinedIcon from '@mui/icons-material/SettingsOutlined';
import BeenhereOutlinedIcon from '@mui/icons-material/BeenhereOutlined';
import { useState } from 'react';
import { ThemeProvider } from '@mui/material';
import { Link } from 'react-router-dom';
import
{
  NotificationsHead, HeaderContiner, TabsContiner, Tab, TabText, theme,
} from './styles';
/**
 * Notifications Header
 * @component
 * The component has function clickTab to set active tab
 * @return {React.Component} - Header of notifications page
 */

function NotificationsHeader() {
  // to set active tab
  const [active, setActive] = useState([1, 0, 0, 0]);
  // fuction to set styling of active tab
  const clickTab = (index) => {
    const arr = [...active];
    arr[index] = 1;
    setActive(arr);
  };
  return (
    <ThemeProvider theme={theme}>
      <HeaderContiner data-testid="notifications-header">
        <NotificationsHead>Notifications</NotificationsHead>
        <TabsContiner>
          <Tab index="1" active={active[0]} onClick={() => { clickTab(0); }}>
            <TabText>Activity</TabText>
          </Tab>
          <Tab index="2" active={active[1]} onClick={() => { clickTab(1); }}>
            <Link to="/messages">
              <TabText data-testid="message-router">Messages</TabText>
            </Link>
          </Tab>
          <Tab active={active[2]} onClick={() => { clickTab(2); }}>
            <BeenhereOutlinedIcon />
            <TabText>Mark as Read</TabText>
          </Tab>
          <Tab active={active[3]} onClick={() => { clickTab(3); }}>
            <SettingsOutlinedIcon />
            <Link to="/">
              <TabText>
                Settings
              </TabText>
            </Link>
          </Tab>
        </TabsContiner>
      </HeaderContiner>
    </ThemeProvider>
  );
}

export default NotificationsHeader;
