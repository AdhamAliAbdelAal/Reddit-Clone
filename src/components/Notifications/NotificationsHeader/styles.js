import styled from '@emotion/styled';
import { createTheme } from '@mui/material';

export const ContinerNotifications = styled('section')(() => ({
  background: '#dae0e6',
  minHeight: '100vh',
}));
export const HeaderContiner = styled('header')(({ theme }) => ({
  height: 131,
  margin: '0 auto',
  width: 648,
  [theme.breakpoints?.down('md')]: {
    width: 'fit-content',
  },
}));
export const NotificationsHead = styled('h1')(() => ({
  padding: ' 40px 0px 21px',
  fontSize: 22,
  fontWeight: '600',
  lineHeight: '26px',
  margin: 0,
}));
export const TabsContiner = styled('nav')(({ theme }) => ({

  display: 'flex',
  [theme.breakpoints?.down('sm')]: {
    flexWrap: 'wrap',
    justifyContent: 'flex-start',
    paddingLeft: '12px',
  },
  [theme.breakpoints?.down('xs')]: {
    justifyContent: 'space-around',
  },
}));
export const TabText = styled('div')(() => ({
  marginLeft: 6,
}));
export const Tab = styled('div')(({ theme, index, active }) => ({
  fontWeight: '700',
  fontSize: '14px',
  color: '#848485',
  marginRight: index === '2' ? 'auto' : '',
  padding: ' 15px 17px 12px',
  cursor: 'pointer',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  borderBottom: index === '1' ? `3px solid ${theme.palette.primary.main}` : '',
  ...((active === 1) && {
    div: {
      color: 'black',
    },
  }),
  '&:hover div': {
    color: 'black',
  },
  '& a': {
    textDecoration: 'none',
    color: 'inherit',
  },
  [theme.breakpoints?.down('sm')]: {
    marginRight: index === '2' ? '0px' : '',
  },
  [theme.breakpoints?.down('xs')]: {
    padding: '15px 14px 12px',
  },
}));

export const theme = createTheme({
  breakpoints: {
    values: {
      md: 670,
      sm: 500,
      xs: 386,
    },
  },
});
