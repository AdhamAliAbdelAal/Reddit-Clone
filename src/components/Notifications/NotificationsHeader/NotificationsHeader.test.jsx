import {
  render, screen, fireEvent,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../../../App';
import NotificationsHeader from './NotificationsHeader';

// render notifications page
it('should render notifications page', () => {
  window.history.pushState({}, '', '/notifications');
  render(<App />);
  expect(screen.getByTestId('notifications')).toBeInTheDocument();
});

// render notifications Header
it('should render notifications page', () => {
  window.history.pushState({}, '', '/notifications');
  render(<App />);
  expect(screen.getByTestId('notifications-header')).toBeInTheDocument();
});

// message page by click message button
it('routes to message page by click message button', async () => {
  createMemoryHistory({ initialEntries: ['/notifications'] });
  render(<App />);
  const notifications = screen.getByTestId('notifications-header');
  fireEvent.click(screen.getByTestId('message-router'));
  expect(notifications).not.toBeInTheDocument();
});

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<Router><NotificationsHeader /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
