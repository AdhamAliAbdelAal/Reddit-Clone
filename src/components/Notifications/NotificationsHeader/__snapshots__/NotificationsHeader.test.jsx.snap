// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<header
  className="css-1smv5mi"
  data-testid="notifications-header-test"
>
  <h1
    className="css-z0evd0"
  >
    Notifications
  </h1>
  <nav
    className="css-1em3x1k"
  >
    <div
      className="css-1rudzjd"
      onClick={[Function]}
    >
      <div
        className="css-m75dnw"
      >
        Activity
      </div>
    </div>
    <div
      className="css-o8qxya"
      onClick={[Function]}
    >
      <a
        href="/messages"
        onClick={[Function]}
      >
        <div
          className="css-m75dnw"
          data-testid="message-router"
        >
          Messages
        </div>
      </a>
    </div>
    <div
      className="css-xxc4pc"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="BeenhereOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 1H5c-1.1 0-1.99.9-1.99 2L3 15.93c0 .69.35 1.3.88 1.66L12 23l8.11-5.41c.53-.36.88-.97.88-1.66L21 3c0-1.1-.9-2-2-2zm-7 19.6-7-4.66V3h14v12.93l-7 4.67zm-2.01-7.42-2.58-2.59L6 12l4 4 8-8-1.42-1.42z"
        />
      </svg>
      <div
        className="css-m75dnw"
      >
        Mark as Read
      </div>
    </div>
    <div
      className="css-xxc4pc"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SettingsOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
        />
      </svg>
      <a
        href="/settings"
        onClick={[Function]}
      >
        <div
          className="css-m75dnw"
        >
          Settings
        </div>
      </a>
    </div>
  </nav>
</header>
`;
